compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2027,6 +2033,47 @@ fn default_to_whole_archive(sess: &Session, crate_type: CrateType, cmd: &dyn Lin
         && !(sess.target.limit_rdylib_exports && cmd.exported_symbol_means_used_symbol())
 }
 
+// Write the debugger visualizer files for each crate to the temp directory and gather the file paths.
+fn collect_debugger_visualizers(
+    tmpdir: &Path,
+    sess: &Session,
+    crate_info: &CrateInfo,
+) -> Vec<PathBuf> {
+    let mut visualizer_paths = Vec::new();
+    let debugger_visualizers = &crate_info.debugger_visualizers;
+    let mut index = 0;
+
+    for (cnum, visualizers) in debugger_visualizers {
+        let crate_name = if cnum == &LOCAL_CRATE {
+            crate_info.local_crate_name.as_str()
+        } else {
+            crate_info.crate_name[&cnum].as_str()
+        };
+
+        for visualizer in visualizers {
+            let visualizer_out_file = tmpdir.join(format!("{}-{}.natvis", crate_name, index));
+
+            match fs::write(&visualizer_out_file, visualizer.src.as_slice()) {
+                Ok(()) => {
+                    visualizer_paths.push(visualizer_out_file.clone());
+                    index += 1;
+                }
+                Err(error) => {
+                    sess.warn(
+                        format!(
+                            "Unable to write debugger visualizer file `{}`: {} ",
+                            &visualizer_out_file.display().to_string(),
```

Comment:
You can make this just `visualizer_out_file.display()`. The `display()` method returns an object that implements `std::fmt::Display` and thus can be used in `format!()`. No need to first copy it into a separate `String`.